src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -454,6 +471,10 @@ where
                     // Now we drop the activated output handle to force timely to emit any pending
                     // batch. It's crucial that this happens before our attempt to yield otherwise
                     // the buffer would get stuck in this operator.
+                    if let Some(emit_cap) = emit_cap.take() {
+                        output.session(&emit_cap).give_container(&mut batch);
+                        batch.clear();
+                    }
```

Comment:
Should we assert that the batch is empty here?

Reply:
sure, will add

Reply:
Yes, I'll add it